### DISCUSSION TITLE
Proxy for axios

### DISCUSSION THREAD

**Discussion by lionchi** [chimurai/http-proxy-middleware#534] [Q&A]
**src/setupProxy.ts**
import express from 'express';
import {createProxyMiddleware} from 'http-proxy-middleware';
const app = express();
app.use('/registration-api/**', createProxyMiddleware({target: 'http://localhost:8710', changeOrigin: true}));
app.listen(3000);

**config axios**
export const http = axios.create({
    baseURL: baseUrl()
});

**where baseUrl use for context path**
export function baseUrl() {
    const applicationUrl = document.head.querySelector('meta[name=application-url]');
    const baseUrl = applicationUrl ? applicationUrl.getAttribute('content') : '/';
    return baseUrl == null ? '/' : baseUrl;
}

**api**
export function registration(values: RegistrationFormData): Promise<AxiosResponse<OperationResult>> {
   return http.post('/registration-api/registration', values);
}

**api call result**
POST http://localhost:3000/registration-api/registration 404 (Not Found)

What am I doing wrong ?

**Comment by mikco11** [comment]
I have a similar problem with http-proxy-middleware.
I am using 0.19.1 following official documentation but i could only proxy via configuration in `package.json` but not via `setupProxy.js`

**Comment by AubreyHewes** [comment]
@lionchi 

* Using the proposed (when you opened the issue) [issue template](https://github.com/chimurai/http-proxy-middleware/blob/master/.github/ISSUE_TEMPLATE.md) would help.
  * Formatting your code snippets would help!
  * Adding description of what you actually want to do and why it is not working would help.
  * Adding a description of the frameworks you are using would help.
  * Adding a [minimal repo/online example](https://github.com/chimurai/http-proxy-middleware/blob/master/.github/ISSUE_TEMPLATE.md#reproducible-demo) of your problem would help.

Some assumptions:
 * `src/setupProxy.ts` makes me think of [CRA/create-react-app](https://create-react-app.dev) where typescript is [not supported](https://create-react-app.dev/docs/proxying-api-requests-in-development#configuring-the-proxy-manually) for the `src/setupProxy.js` file (yet).
   * Is a CRA project and so the problem is not with this project.

@Mikco11 

 * See the [issue template](https://github.com/chimurai/http-proxy-middleware/blob/master/.github/ISSUE_TEMPLATE.md) for minimal information required.
    * Adding description of what you actually want to do and why it is not working would help.
    * Adding a description of the frameworks you are using would help.
    * Adding a [minimal repo/online example](https://github.com/chimurai/http-proxy-middleware/blob/master/.github/ISSUE_TEMPLATE.md) of your problem would help.

Some assumptions:
 * Your stated `0.19.1` version, is a version of [axios](https://github.com/axios/axios/releases/tag/v0.19.1), not this [project](https://github.com/chimurai/http-proxy-middleware/releases/tag/v0.19.1) (if it is this project you are using a very old version; advise to update to the latest, asap!).
 * proxy via `package.json` configuration also makes me think of [CRA/create-react-app](https://create-react-app.dev/docs/proxying-api-requests-in-development)
   * Is a CRA project and so the problem is not with this project.
